src/mcfret.c

```diff
@@ -0,0 +1,366 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <time.h>
+#include <fftw3.h>
+#include "omp.h"
+#include "types.h"
+#include "NISE_subs.h"
+#include "mcfret.h"
+#include "project.h"
+
+
+
+
+   
+void mcfret(t_non *non){
+  /* Call the MCFRET Routine */
+   if (!strcmp(non->technique, "MCFRET") || (!strcmp(non->technique, "MCFRET-Autodetect")) || (!strcmp(non->technique, "MCFRET-Absorption"))
+        || (!strcmp(non->technique, "MCFRET-Emission")) || (!strcmp(non->technique, "MCFRET-Coupling")) || (!strcmp(non->technique,
+        "MCFRET-Rate")) || (!strcmp(non->technique, "MCFRET-Analyse")) ) {
+        printf("Perfroming the MCFRET calculation.\n");
+    }
+
+/* Define variables and arrays */
+   /* Integers */
+    int nn2;
+    int itime,N_samples;
+    int samples;
+    int x,ti,tj,i,j;
+    int t1,fft;
+    int elements;
+    int cl,Ncl;
+
+    /* Response functions for emission and absorption: real and imaginary part*/
+    float *re_S_Abs,*im_S_Abs;
+    float *re_S_Emi,*im_S_Emi;
+    /*Hamiltonian of the whole system - all donors and acceptors included*/
+    float *Hamil_i_e;
+    /*Vectors representing time dependent states: real and imaginary part*/
+    float *vecr, *veci;
+    /*Density matrix*/
+    float *dens_mat;
+
+    float shift1;
+  /* 1D Fourier transform */
+    fftw_complex *fftIn,*fftOut;
+    /*was is das?*/
+    fftw_plan fftPlan;  
+
+  /* Time parameters */
+    time_t time_now,time_old,time_0;
+  /* Initialize time */
+    time(&time_now);
+    time(&time_0);
```

Comment:
We initialize the time twice, this seems odd. I don't understand why we cant remove line 54. If it is needed, could you add a comment explaining it?

Reply:
This comment is still relevant here